### DISCUSSION TITLE
Hackday adjacent to NSIDC UWG meeting - who's interested?

### DISCUSSION THREAD

**Discussion by JessicaS11** [earthaccess-dev/earthaccess#794] [General]
Coming up quickly in October, several of us who are typically remote will be in Boulder for the NSIDC User Working Group (UWG) meeting. Anyone interested in adding a half or full day of collaborative hack time before or after the UWG meeting (which is Tues + Wed)?

See also: #775 

My personal thoughts:
- Thursday would be better so travel can happen Mon and Fri and not interfere with the weekend.
- As a non-local, I'm happy to attend but have no idea if there's a conference room (e.g. at NSIDC) we could use. I think BYO coffee and a lunch break are perfectly fine to cut down on logistics/cost planning.
- It would be great to have high level details (day, timing) hammered out by Fri 6 Sept for those of us who need to book flights and other travel arrangements.
- Are there any other UWG members or locals we should tag here who might be interested? @tsutterley

**Comment by andypbarrett** [comment]
@JessicaS11 thanks for starting this discussion.

- I was also thinking Thursday as the day/date
- We can organize a conference room at NSIDC.
- I had not thought about caffeine and other necessary life support items.  I think BYO is fine.  There are plenty of options and we could always order in from one of the local restaurants/cafes.
- I was definitely going to tag @tsutterley and @alekpetty but we could also have an open invitation to other interested UWG members, as well `earthaccess` and `icepyx` adjacent locals.
- I think that in addition to dates and times, it would be good to have a list of candidate projects so that we are not spending too much time figuring out what to do.

**Comment by andypbarrett** [comment]
I've booked two rooms at NSIDC for Thursday 10 October.  We don't have to use both but the large meeting room is in use for an ESDIS meeting and if we have two "hacks" we can have space.

**Comment by andypbarrett** [comment]
@abarciauskas-bgse Mikala Beig mentioned you might be in Boulder around this time.  If you are free on the 10th October it would be great to have to participate.

**Comment by abarciauskas-bgse** [comment]
Ugh I would love to come (thank Mikala for thinking of me!) unfortunately I leave on October 4. I will be around Sept 23-Oct 4, with free time Sept 26-29.

**Comment by tsutterley** [comment]
I added a personal day to my travel request, so I'll be there on the 10th for the _Hackday_ :)

**Comment by betolink** [comment]
Definitely interested, there are many topics that can be hacked by all of us, VirtualiZarr and improvements to remote subsetting operations with cloud-optimized HDF5 files for example.

**Comment by JessicaS11** [comment]
@asteiker and I spent some time brainstorming for this hackday next week. Here are some of the ideas we came up with as topics to explore/hack on:

- icepyx/earthaccess:
  - What earthaccess functions could we swap out under the hood in icepyx, including subsetting query and dataset query
  - Icepyx also has a lot of checks (making sure temporal inputs are in right order, and spatial bounding box is sound when crossing dateline) --> which things should we check for users, and where should those checks live (e.g. does pythonCMR already handle them?)
  - Sliderule is also looking at replacing some search functionality with earthaccess
- Work on earthaccess docs and examples - can you find and access your favorite dataset with earthaccess?
- Adding widgets to browse through results:
  - https://github.com/nsidc/earthaccess/issues/762 
  - Related to explore method? Where is that development happening? https://github.com/betolink/earthaccess ? 
- Creating simplified polygons for search:
  - SlideRule has implemented a robust solution to this
- Expand earthaccess to support geojson/shapfile upload?
- VirtualiZarr

**Reply by mfisher87** [reply]
>Adding widgets to browse through results: 

:star_struck:
